modules/core/src/main/java/org/apache/ignite/spi/communication/tcp/TcpCommunicationMetricsListener.java

```diff
@@ -404,36 +423,50 @@ public static String receivedMessagesByTypeMetricName(Short directType) {
         /**
          * Sent messages count metrics grouped by message node id.
          */
-        public volatile Map<UUID, LongAdderMetric> sentMsgsMetricsByNodeId = new HashMap<>();
+        public volatile Map<UUID, AtomicLong> sentMsgsMetricsByNodeId = new HashMap<>();
 
         /**
          * Received messages metrics count grouped by message node id.
          */
-        public volatile Map<UUID, LongAdderMetric> rcvdMsgsMetricsByNodeId = new HashMap<>();
+        public volatile Map<UUID, AtomicLong> rcvdMsgsMetricsByNodeId = new HashMap<>();
+
+        /**
+         * Sent messages count metrics grouped by message node consistent id.
+         */
+        public volatile Map<Object, LongAdderMetric> sentMsgsMetricsByConsistentId = new HashMap<>();
+
+        /**
+         * Received messages metrics count grouped by message node consistent id.
+         */
+        public volatile Map<Object, LongAdderMetric> rcvdMsgsMetricsByConsistentId = new HashMap<>();
 
 
         /**
          * Collects statistics for message sent by SPI.
-         *
-         * @param msg Sent message.
+         *  @param msg Sent message.
+         * @param consistentId Receiver node consistent id.
          * @param nodeId Receiver node id.
          */
-        private void onMessageSent(Message msg, UUID nodeId) {
+        private void onMessageSent(Message msg, Object consistentId, UUID nodeId) {
```

Comment:
`nodeId` should be removed as it not used.

Reply:
It is used in last line of the method.